vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -392,6 +524,74 @@ def __init__(
 
         self.quant_method.create_weights(layer=self, **moe_quant_params)
 
+        # FIXME: (Yi) we need to wrap the `torch.ops.hpu.mixture_of_experts` as a module,
+        # so that INC can patch it for measurement and quantization.
+        layer = self
+        ep_shift = self.ep_rank * self.num_experts
+        if VLLM_REQUANT_FP8_INC:
```

Comment:
@yiliu30 , need to find a better place to put below codes changes. Can we follow VllmMixtureOfExpertsOp non fp8 way by not change to vllm codes?

Reply:
Agree, I moved most of code to `create_weights`, leaving only the code for creating the `moe_op`.